### ISSUE TITLE
Compliance and finance workflow testing

### ISSUE THREAD

**Issue by abhiXII** [HyphaApp/hypha#3558]
Testing compliance and finance workflow:

- [x] Minor typo on the create determination page: It should be "When the text is first added and then every time it is changed."
<img width="832" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/adcc7d4b-5c6c-4dc9-96a5-c2d845a8a4b0">
There's also an option to select whether to send message to the applicant. so the text and this checkbox seem a bit conflicting
<img width="637" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/0eda3a4e-4331-4091-9ac6-13c56926f21d">

- [x] #3588
<img width="649" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/f6dec3f0-faf5-4276-b75a-6ffde766c37c">
To be discussed with Saurabh in team meet

- [x] There's some changes to this box 
<img width="463" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/c68f0e81-48a7-4324-8eaf-89c875748184">
we changed the text to 'Waiting for' with a blue arrow.

- [x] #3589
<img width="886" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/6c4d9e50-14b5-4837-91fe-dab93974dd6e">
like this
<img width="662" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/57081805-5b50-4ae9-9fcd-a26638c5bff7">

- [x] We don't need huge input fields for phone and email
<img width="890" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/716457ae-fa57-4a32-bcd0-9e7f92570132">
Can we please reduce height like this
<img width="644" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/45cf280e-ad93-45bf-a4e5-155e5789fe0a">

- [x] Can we please recheck all the spacing between fields
<img width="924" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/2da2b987-0133-4421-b059-1452ac6d7166">
<img width="745" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/e6c63cfb-1022-49ae-9579-c1998378669a">

- [x] Are we not planning to add rich text formatting in these fields? I think we did discuss regarding formatting.
<img width="867" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/d106b405-fe39-4cdd-9db3-e728a6b892d6">

- [x] #3590
<img width="687" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/b249c5f9-0c8b-4afe-97ab-19cd6e11a95f">
like this 
<img width="544" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/ae987e83-210d-4a71-bbbc-c3007a88d7b0">

- [x] Can we change Hypha to OTF in the text where we're referring to the organisation. I think it will help if we can also see it on the test site. What do you think @fourthletter @frjo  
<img width="372" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/f2f6bc56-39df-4bfa-8ac0-3270e921d2ba">

- [x] Can we please correct the spacing on the PAF page after it is filled out 
<img width="947" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/af78d645-384a-4c3f-b148-c1db2331eb34">

- [x] The spacing between the text is making it harder to read through the paf page
https://www.figma.com/file/cmrfUklaIibgEYRFUIifuE/Hypha-designs?type=design&node-id=512-13024&mode=design&t=IJhQZuFRHgG3Iq61-4
can we please implement spacing as shown in the design here: [link](https://www.figma.com/file/cmrfUklaIibgEYRFUIifuE/Hypha-designs?type=design&node-id=512-13024&mode=design&t=IJhQZuFRHgG3Iq61-4)
Also check the styling of the budget file, we don't need something so big and blue.

- [x] #3650
<img width="1512" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/d9693d61-ce26-4596-8bae-d3c0e40d16ae">
[like this](https://www.figma.com/file/cmrfUklaIibgEYRFUIifuE/Hypha-designs?type=design&node-id=512-13024&mode=design&t=IJhQZuFRHgG3Iq61-4)

- [x] Should we also add "Assign approver" button here inside the paf page. I think this page should have same controls as the project page?
<img width="1463" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/34f1c69a-65a1-4f53-aff5-c8c7c9a5032c">

- [x] This project is showing some activity history even before the project was created
<img width="1420" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/34fe7b99-8ce2-4987-900a-f74167ba3b15">
https://test-apply.hypha.app/apply/projects/88/#activity-feed

- [x] What does 'Created' mean? like project created? Can we make it a bit more specific like "Committed from submission to project"
<img width="665" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/365df741-dc50-4ce3-8142-e0b4dffe926a">
Should we also capture when someone gets assigned? because there may be changes to it and we can capture that change history? @fourthletter 

- [x] Can we make w8/w9 conditionally mandatory for users — make it mandatory if it's applicable to them. or maybe we make it non-mandatory and add text (if applicable)
<img width="942" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/7b7121e9-24bf-4f24-99cc-75637cf88337">

- [x] When the project contracts are under approval, we can say  "Awaiting contract approval by OTF" or  "Awaiting contract approval from lead" instead of "Awaiting contract approval from Staff"
<img width="736" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/9cbad7de-b676-46f4-9c82-ffd64b55f817">

- [x] When a staff is viewing the invoice, we also need to show the vendor name
<img width="1365" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/2d491f21-578e-468f-b1c8-9cfe428d19b5">
like this
<img width="936" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/a5f08442-a56d-4e8a-ad55-94d6559fc61e">

- [x] We have duplicate invoice number columns now. we can remove the old one or we can tackle it when working on the dashboard
<img width="1299" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/e0053271-1bac-4db1-a94b-36deeb10433e">
And the new invoice number will be clickable

- [x] We also need to add dates to these statuses
<img width="941" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/0e109164-5ada-4c09-a6d3-69454503ff1c">
<img width="531" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/8b66e71c-52a1-4a32-80fd-cfa8fde502ce">
We need the show comment option here too
<img width="717" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/a3c8cbd9-536a-4c73-9da9-8d5a6b37a2a1">
Also, let's change the button "View/Hide status history" to just "View/Hide" so that it occupies less space

- [x] To vendor, can we show 'changes requested by OTF' instead of specific otf staff?
<img width="947" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/7423ba4d-1fe8-4167-95fb-59766efe36e8">

- [x] We need to add more invoice status for Finance: Paid and Payment returned after it is approved, and also Declined when approving it
<img width="627" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/9e123422-3be3-4f09-b43a-6c1434a593fb">

- [x] We had also made some formatting changes to the PAF pdf download like this 
<img width="760" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/35324a35-0620-43ff-9244-15fe8bbb4f6a">

- [x] I got an error while filling out the form 
<img width="800" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/846fc8ab-b5a2-44b2-9f6c-d153dc6047d2">
and now after filling it, I'm not able to submit it
<img width="771" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/f585329f-f21f-46c2-8f64-541249b9b708">
and when I click no save draft it says 
<img width="601" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/0845cd49-6416-413a-b239-c5c40dd2b370">

To be tested: proposal attachments

**Comment by abhiXII** [comment]
We lost the Fill in button here 
<img width="1359" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/e8c7c746-221e-4a3c-93c0-5ca22793f9a5">

**Comment by sandeepsajan0** [comment]
@abhiXII Which user are you trying it with? It should be available for staff.

Update: It is available for user 'Abhishek staff'. Looks like you are trying it with staff+admin role.
<img width="1053" alt="image" src="https://github.com/HyphaApp/hypha/assets/23638629/5f76ab89-319c-420c-86f2-f724db40ce59">

**Comment by frjo** [comment]
@sandeepsajan0 If we use `is_apply_staff` any user with "admin" role should be included.

https://github.com/HyphaApp/hypha/blob/4c4d2d329399930b73f24f9f980a09e64f24a52c/hypha/apply/users/models.py#L243-L245

**Comment by abhiXII** [comment]
@sandeepsajan0 this is a bit strange when I release it, I generally become the staff user. this time I don't what user I became. it's not even in the list. See, there's no `Abhishek` all with some second name
<img width="536" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/950efb26-5a27-410e-8390-d16a8c359d90">
When I actively changed it to Abhishek Staff, it worked fine.

**Comment by fourthletter** [comment]
>> Can we change Hypha to OTF in the text where we're referring to the organisation. I think it will help if we can also see it on the test site. What do you think @fourthletter @frjo  
<img width="372" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/f2f6bc56-39df-4bfa-8ac0-3270e921d2ba">

My proposed text: 

- "Contract documents pending review"
- "Contract documents have been approved"
- Vendor receives an alert when they need to review contract documents

>> Should we also capture when someone gets assigned? because there may be changes to it and we can capture that change history? @fourthletter 

Yes, we should capture when someone change the 'Lead' in the activity feed. We could also capture when someone has been assigned as a PAF approver.

**Comment by abhiXII** [comment]
Yeah @fourthletter it perhaps works well without the name Hypha/OTF. we could just say "Awaiting project documents to be created. Please check back when the project has moved to contracting stage." 
Since this at the time of PAF creation and approvals.

**Comment by sandeepsajan0** [comment]
> 1. Minor typo on the create determination page: It should be "When the text is first added and then every time it is changed."

We are using StreamField forms for determination so this text and checkbox option can be changed directly via admin as per the requirements.

> 2. Title fonts look different now. probably they lost its bold style

Yes, headings on detail pages are not bold now. It is kind of a general issue, happening for submissions, projects, and all other pages. @theskumar @frjo @abhiXII Do we want its bold style back or ..?  (A separate issue and a separate PR?)

> 4. The sub-text should wrap only after the width of the input box.

It is again a general issue. We have to update the help text css for every Streamfield form. (A separate issue and a separate PR?)

> 5. We don't need huge input fields for phone and email

I think it is just what field type you choose for these fields in wagtail. This can be easily fixed via wagtail admin just by updating the field type.

> 8. Can we make the text in the buttons centered

This again is a general enhancement. We will try to fix it as a separate issue and PR. (A separate issue and a separate PR?)

> 9. Can we change Hypha to OTF in the text where we're referring to the organisation. I think it will help if we can also see it on the test site.

"Hypha" is just an organization name, not hardcoded(picked from env variables) so on the live site it should be "OTF". And if we want to update the text and don't want to include the org name anymore then it is also fairly easy just let me know the text.

> 11. can we please implement spacing as shown in the design. Also, check the styling of the budget file, we don't need something so big and blue. 

Worked on the spacing part(should be on test soon).
<img width="814" alt="image" src="https://github.com/HyphaApp/hypha/assets/23638629/578ee60c-a2e8-44a5-aede-d765e1c95532">

Regarding budget file styling, hypha doesn't know about the budget it is just another file field for the code, and file field styling is the same everywhere. So, It needs discussion if we really want it like that.


> 12. Seems like we lost the grey background of the PAF page (and other pages also)

Yes, We have removed the grey background for all the pages. Now, we are thinking of having it back for dashboards(latest designs) because it contains multiple sections but no plan for the detail pages yet right @theskumar @frjo?


> 13. Should we also add "Assign approver" button here inside the paf page. I think this page should have same controls as the project page?

Fixed, The 'assign approver' button is available there on the paf detail page.
<img width="1269" alt="image" src="https://github.com/HyphaApp/hypha/assets/23638629/13048be0-4f11-401f-811d-7dd2d5d3082b">
 
 > 14. This project is showing some activity history even before the project was created
 
We have already discussed this, it can be a glitch because of db reset on the test. As it happens only once so can leave it for now.

> 15. What does 'Created' mean? like project created? 
capture when someone gets assigned?

Fixed, it has been updated to "Created project".
<img width="1269" alt="image" src="https://github.com/HyphaApp/hypha/assets/23638629/7997d09e-ee55-4570-8891-bc28a40d4431">
Also captured when someone gets assigned to a PAF.
<img width="1269" alt="image" src="https://github.com/HyphaApp/hypha/assets/23638629/904dba08-bc11-4212-805f-0aea5b5002fb">

> 16. Can we make w8/w9 conditionally mandatory for users

These are either project documents or contracting documents and both of these are configurable via admin. We can set it as required or optional just via admin.

> 17. When the project contracts are under approval, we can say "Awaiting contract approval by OTF"

It depends on which user groups are we using. If you see it as an Applicant it will show "OTF" or whatever is configured org short name. For contracting and other employees, it shows "Staff".

> 18. When a staff is viewing the invoice, we also need to show the vendor name

Already added the vendor field there, but I'll check if there is any issue with showing the name.

> 19. We have duplicate invoice number columns now. we can remove the old one

Fixed, There are only new invoice numbers now.

> 21. To vendor, can we show 'changes requested by OTF' instead of specific otf staff?

Fixed, and updated staff name to org short name.

> 22. We need to add more invoice status for Finance: Paid and Payment returned after it is approved, and also Declined when approving it

Fixed in [this PR](https://github.com/HyphaApp/hypha/pull/3578) but having some confiliting issues so couldn't put it to test yet. I think we can test it after finishing/merging the other changes?

> 23. We had also made some formatting changes to the PAF pdf download like this

Fixed, not exact design but almost similar. Actually, it is hard to work with pdf css, it supports very limited methods or properties.

> 24. I got an error while filling out the form

I think it is related to submission form and not sure but I think we have talked about this?

**Comment by abhiXII** [comment]
Thanks Sandeep. 
Yea, I think all the general submissions tackled separately would also work.
I have checkmarked above as per your update

> 15. Also captured when someone gets assigned to a PAF.
It seems that person who is changing the assignee is showing up in activity feed. I changed the finance approver to Bernard tyers 
<img width="1300" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/9ce05856-3b3c-4f93-95d8-5fd7c4e2f4a7">
but it shows Abhishek Finance
<img width="1287" alt="image" src="https://github.com/HyphaApp/hypha/assets/124244361/6e253162-dd40-485e-9083-db3aba254647">

**Comment by frjo** [comment]
@abhiXII I have changed the short org name on the test site from "Hypha" to "ACME" now. As Sandeep mentioned, this is a setting that each organisation set to whatever they like, On OTFs live site is of course is set to "OTF".

**Comment by frjo** [comment]
@abhiXII @sandeepsajan0 

> Yes, We have removed the grey background for all the pages. Now, we are thinking of having it back for dashboards(latest designs) because it contains multiple sections but no plan for the detail pages yet right @theskumar @frjo?

Add the grey background back wherever it make sense/works well. We discussed that this was likely on pages with a lot of boxes while pages with mostly text worked better with white.

**Comment by fourthletter** [comment]
>> Can we make w8/w9 conditionally mandatory for users — make it mandatory if it's applicable to them. or maybe we make it non-mandatory and add text (if applicable)image

It is possible to make these form mandatory/required or not required in the Wagtail Admin

![image](https://github.com/HyphaApp/hypha/assets/20019656/ef1d04ae-8a7a-4590-9d16-c0256e3e60a7)

**Comment by sandeepsajan0** [comment]
>It seems that person who is changing the assignee is showing up in activity feed. I changed the finance approver to Bernard tyers  but it shows Abhishek Finance

It might be a bug because when I tested it, I assigned it to myself. But a question arises here, what do we want to show, 'assignee', 'assigner', or both?

I noticed one more bug in my testing: 'Approved by staff' and 'Changes requested by finance' are visible to the vendor. Need to hide them because these are internal statuses.
@abhiXII I have created separate issues for general problems so we can tackle them there. And finance and invoice-related stuff has been fixed so you may test for the remaining points and let me know if you find any other issues.

**Comment by sandeepsajan0** [comment]
As we have fixed the grey background for paf issue with #3650. So, we can close it for now. We can open a new issue for upcoming enhancements.
